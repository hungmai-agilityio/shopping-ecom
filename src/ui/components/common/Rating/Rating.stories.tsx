import type { Meta, StoryObj } from '@storybook/react';

// Components
import { Rating } from '@/ui/components';

const meta: Meta<typeof Rating> = {
  component: Rating,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    icon: {
      description: 'Link image for render icon rating',
      control: 'text'
    },
    count: {
      description: 'Number of rating icons to be displayed (max = 5)',
      control: 'number'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Rating>;

export const primary: Story = {
  args: {
    count: 5,
    icon: '/star.svg'
  }
};

export const customIcon: Story = {
  args: {
    count: 5,
    icon: '/heart.svg'
  }
};
