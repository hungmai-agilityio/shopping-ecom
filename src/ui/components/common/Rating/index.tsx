import Image from 'next/image';

interface RatingProps {
  icon?: string;
  count: number;
}

const Rating = ({ icon = '/star.svg', count }: RatingProps) => {
  const renderStars = Array.from({ length: Math.min(count, 5) });

  return (
    <div className="flex gap-1">
      {renderStars.map((_, index) => (
        <Image
          key={index}
          src={icon}
          alt={`rating ${index + 1}`}
          width={20}
          height={20}
          priority
        />
      ))}
    </div>
  );
};

export default Rating;
