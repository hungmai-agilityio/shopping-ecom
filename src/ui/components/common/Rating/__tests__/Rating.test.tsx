import Rating from '@/ui/components/common/Rating';
import { render, screen } from '@testing-library/react';

describe('Rating component', () => {
  test('Should render component with full rating', () => {
    render(<Rating icon="/star.svg" count={5} />);

    const star = screen.getByAltText('rating 5');
    expect(star).toBeInTheDocument();
  });

  test('Do not render the component with an out of bounds rating', () => {
    render(<Rating icon="/star.svg" count={10} />);

    const star = screen.getByAltText('rating 5');
    expect(star).toBeInTheDocument();
  });

  test('should be render match to snapshot', () => {
    const { container } = render(<Rating icon="/star.svg" count={5} />);

    expect(container).toMatchSnapshot();
  });
});
