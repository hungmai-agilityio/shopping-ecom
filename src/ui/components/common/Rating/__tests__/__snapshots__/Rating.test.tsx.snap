// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component should be render match to snapshot 1`] = `
<div>
  <div
    class="flex gap-1"
  >
    <img
      alt="rating 1"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="20"
      src="/star.svg"
      style="color: transparent;"
      width="20"
    />
    <img
      alt="rating 2"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="20"
      src="/star.svg"
      style="color: transparent;"
      width="20"
    />
    <img
      alt="rating 3"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="20"
      src="/star.svg"
      style="color: transparent;"
      width="20"
    />
    <img
      alt="rating 4"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="20"
      src="/star.svg"
      style="color: transparent;"
      width="20"
    />
    <img
      alt="rating 5"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="20"
      src="/star.svg"
      style="color: transparent;"
      width="20"
    />
  </div>
</div>
`;
